x/wasm/IBC.md

```diff
@@ -178,96 +178,98 @@ type IBCCloseChannel struct {
+}
 
-	// block height after which the packet times out 
-	TimeoutHeight uint64
-	// block timestamp (in nanoseconds) after which the packet times out
-	TimeoutTimestamp uint64
+// Response types
+
+
+type IBCPacketReceiveResponse struct {
+  // Acknowledgement contains the data to acknowledge the ibc packet execution
+  Acknowledgement []byte `json:"acknowledgement"`
+  // Messages comes directly from the contract and is it's request for action
+  Messages []sdk.Msg `json:"messages,omitempty"`
+  // Log contains event attributes to expose over abci interface
+  Log []wasmTypes.LogAttribute `json:"log,omitempty"`
 }
-```
 
-Channel Lifecycle:
+type IBCPacketAcknowledgementResponse struct {
+  Messages []sdk.Msg `json:"messages"`
+  Log []wasmTypes.LogAttribute `json:"log"`
+}
 
-```go
-package lifecycle
-
-// if this returns error, we reject the channel opening
-// otherwise response has the versions we accept
-//
-// It is provided the full ChannelInfo being proposed to do any needed checks
-func OnChanNegotiate(ctx sdk.Context, k *wasm.Keeper, contractAddress sdk.AccAddress, request ChannelInfo) 
-    (version Version, err error) {}
-
-// This is called with the full ChannelInfo once the other side has agreed.
-// An error here will still abort the handshake process. (so you can double check the order/version here)
-//
-// The main purpose is to allow the contract to set up any internal state knowing the channel was established,
-// and keep a registry with that ChannelID
-func OnChanOpened(ctx sdk.Context, k *wasm.Keeper, contractAddress sdk.AccAddress, request ChannelInfo) error {}
-
-// This is called when the channel is closed for any reason
-// TODO: any meaning to return an error here? we cannot abort closing a channel
-func OnChanClosed(ctx sdk.Context, k *wasm.Keeper, contractAddress sdk.AccAddress, request ChannelClosedInfo) error {}
-
-type ChannelInfo struct {
-    // key info to enforce (error if not what is expected)
-    Order channeltypes.Order
-    // The proposed version. This may come from the relayer, from the initiating contract, or the responding contract.
-    // In any case, this is the currently agreed upon version to use and if there is disagreement, the contract should
-    // propose another one (if possible in return value), or return an error
-    ProposedVersion Version
-    // local id for the Channel that is being initiated
-    ChannelID string
-    // these two are taken from channeltypes.Counterparty
-    RemotePortID string
-    RemoteChannelID string
+type IBCPacketTimeoutResponse struct {
+  Messages []sdk.Msg `json:"messages"`
+  Log []wasmTypes.LogAttribute `json:"log"`
 }
 
-type ChannelClosedInfo struct {
-    // local id for the Channel that is being shut down
-    ChannelID string
+type IBCChannelOpenResponse struct {
+  // Result contains a boolean if the channel would be accepted
+  Result                       bool   `json:"result"`
```

Comment:
I would use `Accepted` or `Approved` here. Result implies something more than a boolean, with real data

Reply:
Maybe:

```
struct {
  Success bool
  Error string
}
```

Where Error is only set if Success is false